spec/02-integration/03-admin_api/09-targets_routes_spec.lua

```diff
@@ -298,12 +299,24 @@ describe("Admin API", function()
           upstream_id = upstream3.id,
         })
 
-        -- and an insert of a separate active target
+        -- an insert of a separate active target
         assert(helpers.dao.targets:insert {
           target = "api-3:80",
           weight = 10,
           upstream_id = upstream3.id,
         })
+
+        -- two target inserts (we should only see the last one)
+        assert(helpers.dao.targets:insert {
+          target = "api-4:80",
+          weight = 10,
+          upstream_id = upstream3.id,
+        })
+        api4 = assert(helpers.dao.targets:insert {
+          target = "api-4:80",
+          weight = 10,
+          upstream_id = upstream3.id,
+        })
```

Comment:
a case where there are multiple entries, but the last one has `weight=0` (and hence is inactive) would be nice here.